tests/unit_tests/test_unit.py

```
@@ -555,31 +542,49 @@ def test_setup_model(monkeypatch):
+def test_get_model_weights(monkeypatch, selector, expected_filename):
     test_releases = ["3.0.0", "3.0.999", "3.999.999"]
     mock_get = MockResponseGet()
     mock_github = functools.partial(MockGithub, test_releases)
 
+    # Matching versions: exact, same minor, same major.
     for version in test_releases:
         with (
             monkeypatch.context() as mnk,
             tempfile.TemporaryDirectory() as tmp_dir,
         ):
+            version = tuple(
```

Ruff (0.15.14)

[warning] `for` loop variable `version` overwritten by assignment target

(PLW2901)